src/rgw/rgw_op.h

```diff
@@ -1703,11 +1703,15 @@ static inline void rgw_get_request_metadata(CephContext *cct,
   }
 } /* rgw_get_request_metadata */
 
-static inline void encode_delete_at_attr(ceph::real_time delete_at,
+static inline void encode_delete_at_attr(boost::optional<ceph::real_time>& delete_at,
 					map<string, bufferlist>& attrs)
 {
-  if (real_clock::is_zero(delete_at)) {
-    return;
+  map<string, bufferlist>::iterator iter = attrs.find(RGW_ATTR_DELETE_AT);
+  if (delete_at == boost::none) {
```

Comment:
Hmm, I wonder whether we really need to have the `RGW_ATTR_DELETE_AT` set for all objects (including those created through the S3 API). Maybe the check solely for `boost::none` would be enough? What do you think?